connectors/sources/tests/test_generic_database.py

```diff
@@ -130,37 +130,40 @@ def test_get_configuration(patch_logger):
     assert config["host"] == "127.0.0.1"
 
 
-def test_validate_configuration_missing_fields(patch_logger):
-    """Test _validate_configuration method check missing fields"""
+@pytest.mark.asyncio
+async def test_validate_config_missing_fields(patch_logger):
```

Comment:
IMO this a very good example, where we can use `@pytest.parametrize` to test all fields, which are not allowed to be empty.